dev/dll/XamlControlsResources.cpp

```diff
@@ -30,65 +47,51 @@ ThemeResources::ThemeResources()
-                }
-                else if (isRS4OrHigher)
-                {
-                    return L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs4_themeresources.xaml";
-                }
-                else if (isRS3OrHigher)
-                {
-                    return L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs3_themeresources.xaml";
-                }
-                else if (isRS2OrHigher)
-                {
-                    return L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs2_themeresources.xaml";
-                }
-                else
-                {
-                    return L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs1_themeresources.xaml";
-                }
+                packagePrefix = L"ms-appx://" MUXCONTROLS_PACKAGE_NAME "/" MUXCONTROLSROOT_NAMESPACE_STR  "/Themes/";
+            }
+
+            hstring releasePrefix;
+
+            if (is19H1OrHigher)
+            {
+                releasePrefix = L"19h1_";
+            }
+            else if (isRS5OrHigher)
+            {
+                releasePrefix = L"rs5_";
+            }
+            else if (isRS4OrHigher)
+            {
+                releasePrefix = L"rs4_";
+            }
+            else if (isRS3OrHigher)
+            {
+                releasePrefix = L"rs3_";
+            }
+            else if (isRS2OrHigher)
+            {
+                releasePrefix = L"rs2_";
             }
             else
             {
-                if (is19H1OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/19h1_themeresources.xaml";
-                }
-                if (isRS5OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs5_themeresources.xaml";
-                }
-                else if (isRS4OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs4_themeresources.xaml";
-                }
-                else if (isRS3OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs3_themeresources.xaml";
-                }
-                else if (isRS2OrHigher)
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs2_themeresources.xaml";
-                }
-                else
-                {
-                    return L"ms-appx:///" MUXCONTROLSROOT_NAMESPACE_STR "/Themes/rs1_themeresources.xaml";
-                }
+                releasePrefix = L"rs1_";
             }
+
+            return packagePrefix + releasePrefix + compactPrefix + postfix;
         }()
     };
 
+    //  Prior to RS5, when ResourceDictionary.Source property is changed, we forgot to clear ThemeDictionaries.
```

Comment:
```suggestion
    // Because of Compact, UpdateSource may be executed twice, but there is a bug in XAML and manually clear theme dictionaries here:
    //  Prior to RS5, when ResourceDictionary.Source property is changed, XAML forgot to clear ThemeDictionaries.
```